src/CameraGstreamer.cpp

```diff
@@ -219,25 +229,43 @@ GstFlowReturn CameraGstreamer::onNewSample( GstElement *element, gpointer user_d
     return GST_FLOW_OK;
 }
 
+void CameraGstreamer::restartPipeline() {
+    std::cout << "[camera "<<_cameraId<<"] Restarting pipeline." << std::endl;
+    stopPipeline();
+    playGetVideoPackets();
+    _restartPipeline = false;
+    std::cout << "[camera "<<_cameraId<<"] Finished restarting pipeline." << std::endl;
+}
+
+microseconds CameraGstreamer::timeSinceLastRestartPipelineUS() const {
+    return DurationUS(std::chrono::steady_clock::now() - _lastPipelineRestartTimestamp).count();
+}
+
+bool CameraGstreamer::shouldRestartPipelineFromNoSignal() const {
+    return ( _noSignal && (timeSinceLastRestartPipelineUS() >= RESTART_PIPELINE_TIMEOUT_IN_US) );
+}
+
 void CameraGstreamer::alignToFps()
 {
     _threadAlignToFpsId = syscall( SYS_gettid );
     const microseconds timeBetweenFramesInUS = 1'000'000 / _framesPerSecond;
-    constexpr microseconds NO_SIGNAL_TIMEOUT_IN_US = 300'000;
     _lastCapturedTimestamp = std::chrono::steady_clock::now();
+
     while ( _isRunning ) {
         const microseconds curr_duration = DurationUS(std::chrono::steady_clock::now() -  _lastCapturedTimestamp).count();
         if( _restartPipeline ) {
```

Comment:
restartPipeline is never true

Reply:
Why?
What if we get EOS from the gstreamer? Then _restartPipeline is supposed to be true.